src/java/org/apache/cassandra/service/snapshot/TableSnapshot.java

```diff
@@ -113,54 +136,174 @@ public long computeSizeOnDiskBytes()
 
     public long computeTrueSizeBytes()
     {
-        return snapshotDirs.stream().mapToLong(trueDiskSizeComputer::apply).sum();
+        DirectorySizeCalculator visitor = new SnapshotTrueSizeCalculator();
+
+        for (File snapshotDir : snapshotDirs)
+        {
+            try
+            {
+                Files.walkFileTree(snapshotDir.toPath(), visitor);
+            }
+            catch (IOException e)
+            {
+                logger.error("Could not calculate the size of {}.", snapshotDir, e);
```

Comment:
same as above.

Reply:
This is working as expected: https://www.slf4j.org/faq.html#paramException

The first parameter will be added to the message, and the exception trace will be print.